### PR TITLE
Fixed support of provided custom work dir in path resolving.

### LINKED ISSUE #26: Mod Not Working
This mod is not working on CLEO 5, tested on alpha 25.

https://www.mixmods.com.br/2018/01/elm-emergency-lights-mod-v1-0-by-nyolc8/

### PR DESCRIPTION
**Pull request by MiranDMC**
Fixes #26

### CODE REVIEW

File: source/CScriptEngine.cpp [hunk 1]
```diff
@@ -737,10 +740,10 @@ namespace CLEO
     {
         std::ostringstream ss;
 
-        auto threadName = GetName();
+        auto threadName = std::string(GetName(), GetName() + 8); // thread name might not be null terminated
```
Comment (x87) [inline source/CScriptEngine.cpp:743]: what about other places where we expect a null-terminated string?

e.g. https://github.com/cleolibrary/CLEO5/blob/master/source/CCustomOpcodeSystem.cpp#L1432

Comment (MiranDMC) [reply source/CScriptEngine.cpp:743]: Right. Perhaps GetName() function should always return c-string.

File: cleo_sdk/CLEO.h [hunk 2]
```diff
@@ -288,7 +288,7 @@ struct CRunningScript
 	bool IsExternal() const { return bIsExternal; }
 	bool IsMission() const { return bIsMission; }
 	bool IsCustom() const { return bIsCustom; } // is this CLEO Script?
-	const char* GetName() const { return Name; }
+	std::string GetName() const { auto str = std::string(Name, Name + 8); str.resize(strlen(str.c_str())); return str; } // make sure it is always null terminated
```
Comment (x87) [inline cleo_sdk/CLEO.h:291]: can it break the 0AAA opcode?

Comment (MiranDMC) [reply cleo_sdk/CLEO.h:291]: Seems it did in case of threads named with 8 characters.

Comment (x87) [reply cleo_sdk/CLEO.h:291]: I remember @OrionSR brought this issue some time ago. Perhaps we should slow down and think more about desired behavior.

Maybe we should keep all the current logic and instead enforce the 8th byte of the name to be a null-terminator?

Comment (MiranDMC) [reply cleo_sdk/CLEO.h:291]: I made GetName returning std::string. So even 8 characters will be packed into string and null terminated.
Opcode 03A3 (set script name) supports up to 8 characters and apparently not includes terminator in that case.

**Pull request merged**